tests/test_smbconf_samba.py

```diff
@@ -0,0 +1,189 @@
+#
+# sambacc: a samba container configuration tool
+# Copyright (C) 2023  John Mulligan
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+
+import pytest
+
+import sambacc.smbconf_api
+import sambacc.smbconf_samba
+
+smb_conf_reg_stub = """
+[global]
+cache directory = {path}
+state directory = {path}
+private dir = {path}
+include = registry
+"""
+
+smb_conf_sample = """
+[global]
+  realm = my.kingdom.fora.horse
+
+[share_a]
+  path = /foo/bar/baz
+  read only = no
+[share_b]
+  path = /foo/x/b
+  read only = no
+[share_c]
+  path = /foo/x/c
+  read only = no
+[share_d]
+  path = /foo/x/d
+  read only = no
+[share_e]
+  path = /foo/x/e
+  read only = no
+"""
+
+
+def _import_probe():
+    try:
+        import samba.smbconf  # type: ignore
+        import samba.samba3.smbconf  # type: ignore # noqa
+    except ImportError:
+        pytest.skip("unable to load samba smbconf modules")
+
+
+def _smb_data(path, smb_conf_text):
+    data_path = path / "_samba"
+    data_path.mkdir()
+    smb_conf_path = path / "smb.conf"
+    with open(smb_conf_path, "w") as fh:
+        fh.write(smb_conf_text.format(path=data_path))
+    return smb_conf_path
+
+
+@pytest.fixture(scope="session")
+def smbconf_reg_once(tmp_path_factory):
+    _import_probe()
+    tmp_path = tmp_path_factory.mktemp("smb_reg")
+    smb_conf_path = _smb_data(tmp_path, smb_conf_reg_stub)
+
+    return sambacc.smbconf_samba.SMBConf.from_registry(str(smb_conf_path))
```

Comment:
Is the `str` of `smb_conf_path` really needed?

Reply:
Yes. Another reason I don't use pathlib much in the library code (as opposed to tests) is that functions provided by Samba do not take inputs other than strings, they're implemented in simple(r) C code and don't support anything else.